var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -20,11 +20,7 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
     periodic, material, crystal, and biological systems
     """
 
-    build_system(
-        conditional("cmake", when="@master:"),
-        conditional("makefile", when="@:2023.1"),
-        default="makefile",
-    )
+    build_system(conditional("cmake", when="@2023.2:"), "makefile", default="makefile")
```

Comment:
I'm getting `build_system=makefile` for `2023.2`:

```
$ spack spec cp2k
Input spec
--------------------------------
 -   cp2k

Concretized
--------------------------------
 -   cp2k@2023.1%gcc@10.3.1~cosma~cuda~elpa~enable_regtests+libint~libvori+libxc+mpi+openmp~pexsi~plumed~pytorch~quip~rocm~sirius~spglib~spla build_system=makefile lmax=5 smm=libxsmm arch=linux-rhel8-broadwell
 ...
 ```

If I restore the conditional for `makefile`, then I get `build_system=cmake`.

Reply:
Also, since the preference is for the latest version, shouldn't the default be `cmake`?

Reply:
This is explained in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#multiple-build-systems.

Reply:
@tldahlgren thanks for the review. Using `makefile` for the latest version as well is the intended behavior. The rationale is to avoid changing the default until CMake picks up more momentum/there is more consensus in the community (which is difficult to judge).

@mtaillefumier and I use CMake all the time, but since we work at the same institution I did not want to "force" this change and therefore reverted to using `makefile` also for the latest version.

We could go for
```bash
 build_system(
        conditional("cmake", when="@master:"),
        conditional("makefile", when="@:2023.2"),
        default="cmake",
    )
```
but with the risk that we will need another PR like this one if we forget to change the versions here at the next release.

Reply:
Are you saying `cp2k@2023.2` can build with `makefile`?

```
$ spack spec cp2k
Input spec
--------------------------------
 -   cp2k

Concretized
--------------------------------
 -   cp2k@2023.2%gcc@10.3.1~cosma~cuda~elpa~enable_regtests+libint~libvori+libxc+mpi+openmp~pexsi~plumed~pytorch~quip~rocm~sirius~spglib~spla build_system=makefile lmax=5 smm=libxsmm arch=linux-rhel8-broadwell
[+]      ^intel-oneapi-mkl@2023.1.0%gcc@10.3.1~cluster+envmods~ilp64+shared build_system=generic threads=none arch=linux-rhel8-broadwell
[+]          ^intel-tbb@2020.3%gcc@10.3.1+shared+tm build_system=makefile cxxstd=default patches=62ba015,ce1fb16,d62cb66 arch=linux-rhel8-broadwell
 -       ^libint@2.6.0%gcc@10.3.1~debug~fma+fortran build_system=autotools tune=cp2k-lmax-5 arch=linux-rhel8-broadwell
[+]          ^autoconf@2.69%gcc@10.3.1 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-rhel8-broadwell
[+]              ^m4@1.4.19%gcc@10.3.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rhel8-broadwell
[+]                  ^libsigsegv@2.14%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^automake@1.16.5%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^boost@1.77.0%gcc@10.3.1+atomic+chrono~clanglibcpp~container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json+locale+log+math~mpi+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout+wave build_system=generic cxxstd=14 patches=4849671,a440f96 visibility=hidden arch=linux-rhel8-broadwell
[+]              ^zstd@1.5.4%gcc@10.3.1~ipo+programs build_system=cmake build_type=RelWithDebInfo compression=none generator=make libs=shared,static arch=linux-rhel8-broadwell
[+]          ^gmp@6.2.1%gcc@10.3.1+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=linux-rhel8-broadwell
[+]          ^libtool@2.4.7%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
 -       ^libxc@6.2.2%gcc@10.3.1~cuda~kxc~lxc+shared build_system=autotools arch=linux-rhel8-broadwell
[+]          ^perl@5.36.0%gcc@10.3.1+cpanm+open+shared+threads build_system=generic arch=linux-rhel8-broadwell
[+]              ^berkeley-db@18.1.40%gcc@10.3.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rhel8-broadwell
 -       ^libxsmm@1.17%gcc@10.3.1~debug~generator~header-only~large_jit_buffer~shared blas=default build_system=makefile arch=linux-rhel8-broadwell
 -           ^binutils@2.40%gcc@10.3.1+gas~gold~gprofng~headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-rhel8-broadwell
[+]              ^diffutils@3.9%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]              ^texinfo@7.0.3%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]      ^netlib-scalapack@2.2.0%gcc@10.3.1~ipo~pic+shared build_system=cmake build_type=Release generator=make patches=072b006,1c9ce5f,244a9aa arch=linux-rhel8-broadwell
[+]          ^cmake@3.26.3%gcc@10.3.1~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-rhel8-broadwell
[+]          ^gmake@4.4.1%gcc@10.3.1~guile build_system=autotools arch=linux-rhel8-broadwell
[+]          ^openblas@0.3.23%gcc@10.3.1~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-rhel8-broadwell
[+]      ^openmpi@4.1.5%gcc@10.3.1~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-rhel8-broadwell
[+]          ^hwloc@2.9.1%gcc@10.3.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rhel8-broadwell
[+]              ^libpciaccess@0.17%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]                  ^util-macros@1.19.3%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]              ^libxml2@2.10.3%gcc@10.3.1~python build_system=autotools arch=linux-rhel8-broadwell
[+]          ^numactl@2.0.14%gcc@10.3.1 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rhel8-broadwell
[+]          ^openssh@9.3p1%gcc@10.3.1+gssapi build_system=autotools arch=linux-rhel8-broadwell
[+]              ^krb5@1.20.1%gcc@10.3.1+shared build_system=autotools arch=linux-rhel8-broadwell
[+]                  ^bison@3.8.2%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]              ^libedit@3.1-20210216%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^pmix@4.2.3%gcc@10.3.1~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=linux-rhel8-broadwell
[+]              ^libevent@2.1.12%gcc@10.3.1+openssl build_system=autotools arch=linux-rhel8-broadwell
[+]          ^zlib@1.2.13%gcc@10.3.1+optimize+pic+shared build_system=makefile arch=linux-rhel8-broadwell
[+]      ^pkgconf@1.8.0%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
 -       ^py-fypp@3.1%gcc@10.3.1 build_system=python_pip arch=linux-rhel8-broadwell
[+]          ^py-pip@23.0%gcc@10.3.1 build_system=generic arch=linux-rhel8-broadwell
[+]          ^py-setuptools@67.6.0%gcc@10.3.1 build_system=generic arch=linux-rhel8-broadwell
[+]          ^py-wheel@0.37.1%gcc@10.3.1 build_system=generic arch=linux-rhel8-broadwell
[+]      ^python@3.10.10%gcc@10.3.1+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-rhel8-broadwell
[+]          ^bzip2@1.0.8%gcc@10.3.1~debug~pic+shared build_system=generic arch=linux-rhel8-broadwell
[+]          ^expat@2.5.0%gcc@10.3.1+libbsd build_system=autotools arch=linux-rhel8-broadwell
[+]              ^libbsd@0.11.7%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]                  ^libmd@1.0.4%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^gdbm@1.23%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^gettext@0.21.1%gcc@10.3.1+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-rhel8-broadwell
[+]              ^libiconv@1.17%gcc@10.3.1 build_system=autotools libs=shared,static arch=linux-rhel8-broadwell
[+]              ^tar@1.34%gcc@10.3.1 build_system=autotools zip=pigz arch=linux-rhel8-broadwell
[+]                  ^pigz@2.7%gcc@10.3.1 build_system=makefile arch=linux-rhel8-broadwell
[+]          ^libffi@3.4.4%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^libxcrypt@4.4.33%gcc@10.3.1~obsolete_api build_system=autotools arch=linux-rhel8-broadwell
[+]          ^ncurses@6.4%gcc@10.3.1~symlinks+termlib abi=none build_system=autotools arch=linux-rhel8-broadwell
[+]          ^openssl@1.1.1t%gcc@10.3.1~docs~shared build_system=generic certs=mozilla arch=linux-rhel8-broadwell
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@10.3.1 build_system=generic arch=linux-rhel8-broadwell
[+]          ^readline@8.2%gcc@10.3.1 build_system=autotools patches=bbf97f1 arch=linux-rhel8-broadwell
[+]          ^sqlite@3.40.1%gcc@10.3.1+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rhel8-broadwell
[+]          ^util-linux-uuid@2.38.1%gcc@10.3.1 build_system=autotools arch=linux-rhel8-broadwell
[+]          ^xz@5.4.1%gcc@10.3.1~pic build_system=autotools libs=shared,static arch=linux-rhel8-broadwell
```

Reply:
From the change at line 23 I was under the impression you wanted it built with `cmake`.

Reply:
Yes, `cp2k@2023.2` can be built with `makefile`. Currently the two build systems coexist. From `cp2k@2023.2` CMake works well (in our experience), but it might be a bit premature to make it the default. Sorry for the confusion.

Reply:
I didn't realize that from the previous version of the directive.  Thanks for clarifying.